templates/ci/python-django.yml

```diff
@@ -0,0 +1,47 @@
+# Python Django
+# Test a Django app on multiple versions of Python.
+# https://aka.ms/yaml
+
+phases:
+
+# =================
+# Test a Django app
+# =================
+- phase: 'DjangoApp'
```

Comment:
Is there some kind of artifact we can produce here?  Even if it's just copying all the source files, that will make it easier for a user to chain into a release pipeline, right?

Reply:
tl;dr this isn't really possible. The best we can do is robocopy the project directory, which I'll do but may not be very satisfactory.

A Django app isn't deployable to production by default. That's intentional by the Django team as far as I can tell. There are a zillion different ways to get it deployment-ready depending on your target environment. You typically have to define your own process for switching between dev/prod settings, a way to plumb secrets in securely, and a way to serve static files and user-generated content.

One specific thing: I can't call `collectstatic` because it depends on a Django setting for where to lay down the collected files. Django settings are defined in code and read-only at runtime, including `manage.py`. Essentially what we'd have to do is provide a Django app which re-implements `collectstatic` but with a command-line override for where to place the files. It's a lot of work and not really worth the investment, given that we aren't solving the end-to-end deployment problem. (I might go open a PR with the Django folks to provide such a feature natively, though.)

Reply:
FWIW, I set up a "deploy to app service" build that did:
* clone
* install Python from nuget directly into source directory
* install dependencies into that Python
* run collectstatic/other Django commands
* zip up entire source directory and publish that

Then a release definition that pushed the zip file to the app service. Now this was using the Windows packages which are not supported (anymore), but it's about the only viable build output for a Django project.

Real-world apps may have tasks for working on JavaScript (minimization/obfuscation/etc.), but it's hard to put those into a template.

Reply:
Thanks @zooba. I think that makes sense in a deploy task.

My benchmark for build was, "will this succeed after going through the Django tutorial?". Running `collectstatic` fails because the tutorial never has you define `STATIC_ROOT`. (Even if it were defined, it probably wouldn't make sense on the agent unless it were carefully crafted with paths relative to `BASE_DIR`.)